kyuubi-server/src/main/scala/org/apache/kyuubi/engine/ProcBuilder.scala

```diff
@@ -320,9 +321,14 @@ trait ProcBuilder {
       .orElse {
         // 2. get from $KYUUBI_HOME/externals/kyuubi-download/target
         env.get(KYUUBI_HOME).flatMap { p =>
-          val candidates = Paths.get(p, "externals", "kyuubi-download", "target")
-            .toFile.listFiles(homeDirFilter)
-          if (candidates == null) None else candidates.map(_.toPath).headOption
+          Paths.get(p, "externals", "kyuubi-download", "target").toFile
+            .listFiles(homeDirFilter)
```

Comment:
How about pulling `val homeDirFilter` out so that we can overwrite the scala-2.13's SPARK_HOME detection in `SparkProcessBuilder`?

Reply:
Let's try.